sources/packages/backend/apps/workers/src/services/application/application.service.ts

```diff
@@ -109,149 +96,4 @@ export class ApplicationService extends RecordDataModelService<Application> {
-      // Reuse the MSFAA that is still valid and avoid creating a new one.
-      msfaaNumberId = existingValidMSFAANumber.id;
-    } else {
-      // Get previously completed and signed application for the student
-      // to determine if an existing MSFAA is still valid.
-      const previousSignedApplication =
-        await this.getPreviouslySignedApplication(
-          application.student.id,
-          application.currentAssessment.offering.offeringIntensity,
-        );
-
-      let hasValidMSFAANumber = false;
-      if (previousSignedApplication) {
-        const [originalAssessment] =
-          previousSignedApplication.studentAssessments;
-        // checks if the MSFAA number is still valid.
-        hasValidMSFAANumber = this.msfaaNumberService.isMSFAANumberValid(
-          // Previously signed and completed application offering end date in considered the start date.
-          new Date(originalAssessment.offering.studyEndDate),
-          // Start date of the offering of the current application is considered the end date.
-          new Date(originalAssessment.offering.studyStartDate),
-        );
-      }
-
-      if (hasValidMSFAANumber) {
-        // Reuse the MSFAA number.
-        msfaaNumberId = previousSignedApplication.msfaaNumber.id;
-      } else {
-        // Create a new MSFAA number case the previous one is no longer valid.
-        const newMSFAANumber = await this.msfaaNumberService.createMSFAANumber(
-          application.student.id,
-          applicationId,
-          application.currentAssessment.offering.offeringIntensity,
-        );
-        msfaaNumberId = newMSFAANumber.id;
-      }
-    }
-
-    // Associate the MSFAA number with the application.
-    application.msfaaNumber = { id: msfaaNumberId } as MSFAANumber;
-
-    return this.repo.save(application);
-  }
-
-  /**
-   * Gets the application that has an MSFAA signed date.
-   * @param studentId student id to filter the applications.
-   * @param offeringIntensity MSFAA are generated individually for full-time/part-time
-   * applications. The offering intensity is used to differentiate between them.
-   * @returns previous signed application if exists, otherwise null.
-   */
-  async getPreviouslySignedApplication(
-    studentId: number,
-    offeringIntensity: OfferingIntensity,
-  ): Promise<Application> {
-    return this.repo
-      .createQueryBuilder("applications")
-      .select(["applications.id", "msfaaNumbers.id", "offerings.studyEndDate"])
-      .innerJoin("applications.studentAssessments", "assessment")
-      .innerJoin("assessment.offering", "offerings")
-      .innerJoin("applications.msfaaNumber", "msfaaNumbers")
-      .innerJoin("applications.student", "students")
-      .where("applications.applicationStatus = :completedStatus", {
-        completedStatus: ApplicationStatus.Completed,
-      })
-      .andWhere("assessment.triggerType = :triggerType", {
-        triggerType: AssessmentTriggerType.OriginalAssessment,
```

Comment:
I believe that we need to mention/explain/validate this change to @JasonCTang that we are no longer relying on the original assessment offering data to check MSFAA validity.
For instance, with the change, a reassessment of an application that was reusing an existing MSFAA, may en up creating a new MSFAA if the offering start date changes.
Assuming that the application's original assessment had the offering start date of Jan 1st, 2023 and it had an MSFAA valid till Jan 15th, 2023.
If the application is reassessed and the offering change to Jan 31st, 2023, a new MSFAA will be required, right?
I believe that this is the right change, we just need to ensure that it is captured in the ticket and that Jason is aware.